packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,3167 @@
+    _focusNode.removeListener(_handleFocusChange);
+    _internalFocusNode?.dispose();
+    _internalFocusNode = null;
+    super.dispose();
+  }
+
+  @override
+  void didUpdateWidget(MenuItemButton oldWidget) {
+    if (widget.focusNode != oldWidget.focusNode) {
+      _focusNode.removeListener(_handleFocusChange);
+      if (widget.focusNode != null) {
+        _internalFocusNode?.dispose();
+        _internalFocusNode = null;
+      }
+      _createInternalFocusNodeIfNeeded();
+      _focusNode.addListener(_handleFocusChange);
+    }
+    super.didUpdateWidget(oldWidget);
+  }
+
+  void _handleFocusChange() {
+    if (!_focusNode.hasPrimaryFocus) {
+      // Close any child menus of this button's menu.
+      _MenuHandleBase.maybeOf(context)?._closeChildren();
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    // Since we don't want to use the theme style or default style from the
+    // TextButton, we merge the styles, merging them in the right order when
+    // each type of style exists. Each "*StyleOf" function is only called once.
+    final ButtonStyle mergedStyle =
+        widget.style?.merge(widget.themeStyleOf(context)?.merge(widget.defaultStyleOf(context))) ??
+            widget.themeStyleOf(context)?.merge(widget.defaultStyleOf(context)) ??
+            widget.defaultStyleOf(context);
+
+    return TextButton(
+      onPressed: _enabled ? _handleSelect : null,
+      onHover: _enabled ? _handleHover : null,
+      onFocusChange: _enabled ? widget.onFocusChange : null,
+      focusNode: _focusNode,
+      style: mergedStyle,
+      statesController: widget.statesController,
+      clipBehavior: widget.clipBehavior,
+      child: _MenuItemLabel(
+        leadingIcon: widget.leadingIcon,
+        shortcut: widget.shortcut,
+        trailingIcon: widget.trailingIcon,
+        hasSubmenu: false,
+        child: widget.child!,
+      ),
+    );
+  }
+
+  void _createInternalFocusNodeIfNeeded() {
+    if (widget.focusNode == null) {
+      _internalFocusNode = FocusNode();
+      assert(() {
+        if (_internalFocusNode != null) {
+          _internalFocusNode!.debugLabel = '$MenuItemButton(${widget.child})';
+        }
+        return true;
+      }());
+    }
+  }
+
+  void _handleHover(bool hovering) {
+    widget.onHover?.call(hovering);
+    if (hovering) {
+      setState(() {
+        assert(_debugMenuInfo('Requesting focus for $_focusNode from hover'));
+        _focusNode.requestFocus();
+      });
+    }
+  }
+
+  void _handleSelect() {
+    assert(_debugMenuInfo('Selected ${widget.child} menu'));
+    widget.onPressed?.call();
+    MenuController.of(context).closeAll();
+  }
+}
+
+/// A menu button that displays a cascading menu as part of a [MenuBar], as a
+/// standalone widget, or as part of a menu defined by [createMaterialMenu].
+///
+/// This widget represents an item in a [MenuBar] or menu that has a submenu.
+/// Like the leaf [MenuItemButton], it shows a label with an optional leading or
```

Comment:
I like `SubmenuButton` (or `SubMenuButton` - whatever is more correct english). MenuItemButton still feels ambiguous because isn't a SubMenuButton also a MenuItem? But if one of them has a less ambiguous name that's still a big win.